lib/KRadioButton.vue

```diff
@@ -189,9 +213,9 @@
         this.$emit('change', this.isChecked, event);
 
         /**
-         * Used to set `value` to `v-model` when checked
+         * Used to set `buttonValue` to `v-model` when checked
          */
-        this.$emit('input', this.value);
+         this.$emit('input',this.buttonValue || this.value);
```

Comment:
Updating this line to something like this:
`this.$emit('input', this.buttonValue === null ? this.value : this.buttonValue);`
would set the input value to whichever property is not `null`, and giving preference to `buttonValue` if both properties are defined.